c/authService.c

```diff
@@ -173,16 +188,208 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
+  value = regcomp(&regex, regexStr, REG_EXTENDED);
+
+  if (value != 0) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx compiled successfully.");
+  } else {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx compilation error %s.", regexStr);
+  }
+  value = regexec(&regex, url, 0, NULL, 0);
+  char urlCpy[250];
+  strcpy(urlCpy, url);
+  int index = 0;
+  while (urlCpy[index]) { // Capitalize query
+      urlCpy[index] = toupper(urlCpy[index]);
+      index++;
+  }
+  if (instanceID < 0) { // Set instanceID
+    instanceID = 0;
+  }
+  if (value == REG_NOMATCH) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx didn't match.");
+    char * token = strtok(urlCpy, "/");
+    int subUrlIndex = -1;
+    while( token != NULL ) {
+      if (strcmp(rootServiceName, "NULL") == 0)
+      {
+        strcpy(rootServiceName, token);
+      } else {
+        strcpy(subUrl[subUrlIndex], token);
+      }
+      subUrlIndex++;
+      token = strtok(NULL, "/");
+    }
+    strcpy(productCode, "ZLUX");
+    strcpy(type, "core");
+  }
+  else if (!value) {
+    char * token = strtok(urlCpy, "/");
+    int subUrlIndex;
+    subUrlIndex = 0;
+    while( token != NULL ) {
+      switch(subUrlIndex) {
+        case 0:
+          strcpy(productCode, token);
+          break;
+        case 1:
+          strcpy(_p, token);
+          break;
+        case 2:
+          strcpy(pluginID, token);
+          break;
+        case 3:
+          strcpy(_s, token);
+          break;
+        case 4:
+          strcpy(serviceName, token);
+          break;
+        case 5:
+          strcpy(_v, token);
+          break;
+        default:
+          strcpy(subUrl[subUrlIndex-6], token); // subtract 6 from maximum index to begin init subUrl array at 0
+      }
+      
+      subUrlIndex++;
+      token = strtok(NULL, "/");
+    }
+    if ((strcmp(pluginID, "ORG.ZOWE.CONFIGJS") == 0) && (strcmp(serviceName, "DATA") == 0))
+    {
+      strcpy(type, "config");
+      strcpy(pluginID, subUrl[0]);
+      strcpy(scope, subUrl[1]);
+      
+    } else {
+      strcpy(type, "service");
+    }
+    char* ch; 
+    char* chReplace;
+    ch = ".";
+    chReplace = "_";
+    for (index = 0; index <= strlen(pluginID); index++)
+  	{
+  		if (pluginID[index] == *ch)  
+      {
+        pluginID[index] = *chReplace;
+      }
+    }
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx match OK.");
+  }
+  else {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "RegEx match failed.");
+  }
+  strcpy(profileName, makeProfileName(type, 
+    productCode, 
+    instanceID, 
+    pluginID, 
+    rootServiceName,
+    serviceName,
+    method,
+    scope,
+    subUrl));
+  // printf("\n\nFinal query profileName & URL %s - %s\n\n", profileName, url);
+  
+  /* Free memory allocated to the pattern buffer by regcomp() */
+  regfree(&regex);
+  return profileName;
```

Comment:
The variable profileName is allocated on the stack so its lifetime is limited to the enclosing function. If you want to return profileName you have to allocate it with safeMalloc and SLHAlloc